rosetta-source/src/main/rosetta/event-position-type.rosetta

```diff
@@ -42,3 +48,35 @@ type Portfolio: <" A Portfolio represents an aggregation of multiple Positions,
+
+type InventoryRecordBase: <"An individual piece of inventory. This represents a single security.">
+    identifer AssignedIdentifier (1..1) <"Unique identifier for this record. This can be used to uniquely identify a specific piece of inventory.">
+    security Security (1..1) <"The security details.">
+
+type Inventory extends InventoryBase: <"Describes an inventory of securities.">
+
+type InventoryRecord extends InventoryRecordBase: <"Describes an individual piece of inventory.">
+
+type AvailableInventory: <"A data type that can be used to describe the inventory of securities that a party holds. ">
+    party Party (0..*) <"Defines all parties involved for the list of inventory records in this set of inventory. For example, when used to describe securities lending availability, this could hold the sender of the availability, the intended recipient, the beneficial owner(s), the lender (which may differ from the sender as the lender may have the same piece of availability going through multiple agents), an agent or a venue.">
+    partyRole PartyRole (0..*) <"Defines the role(s) that party(ies) may have in relation to the inventory.">
+    availableInventoryRecord AvailableInventoryRecord (0..*) <"An array holding the list of inventory being described. Each element in the inventoryRecord array represents an individual piece of inventory i.e. a security.">
+	
+type AvailableInventoryRecord extends InventoryRecord: <"An individual piece of available inventory. This represents a single security and its associated criteria. The criteria are used to describe any restrictions on the securities.">
+    expirationDateTime zonedDateTime (0..1) <"There may be a set period/time restriction associated to the security.">
+    collateral CollateralProvisions (0..*) <"The type of collateral can often be required when determining if the piece of availability being described is suitable for a party.">
+    partyRole PartyRole (0..*) <"An individual security may be held by several agents. Including the party role at this level allows us to reference the party holding this specific item.">
+    priceQuantity PriceQuantity (0..*) <"Primarily used to hold the quantity of the security. An additional element can be used to hold an optional rate associated to this piece of availability.">
+
+type SecurityAvailability extends AvailableInventory: <"SecurityAvailability describes a list of availability records, with each record being an individual security and its associated criteria. Criteria can include the quantity available, the rate at which the security is available to borrow at, as well as other details that can affect the decision as to whether a party wants to borrow the securities being offered.">
+    messageInformation MessageInformation (0..1) <"Allows details related to the availability messaging to be defined"> 
```

Comment:
If the `messageInformation` attribute is optional in both cases, I would put it down into `AvailableInventory` directly. This means you could model a security availability as an `AvailableInventory` type directly. Less types in the model = good.

Reply:
I’ve moved ```messageInformation``` into ```AvailableInventory```, and as you say, this now means that the ```SecurityAvailability``` type is no longer required. I’ve updated the comments in ```AvailableInventory``` accordingly.